file.go

```diff
@@ -0,0 +1,102 @@
+package fdk
+
+import (
+	"compress/gzip"
+	"encoding/json"
+	"errors"
+	"io"
+	"sync"
+	"sync/atomic"
+)
+
+// File represents a response that is a response body. The runner is in charge
+// of getting the contents to the destination. The metadata will be received.
+type File struct {
+	ContentType string        `json:"content_type"`
+	Encoding    string        `json:"encoding"`
+	Filename    string        `json:"filename"`
+	Contents    io.ReadCloser `json:"-"`
+}
+
+// MarshalJSON marshals the file metadata.
+func (f File) MarshalJSON() ([]byte, error) {
+	type alias File
+	return json.Marshal(alias(f))
+}
+
+// FileGZip writes the
+func FileGZip(filename, contentType string, contents io.ReadCloser) File {
```

Comment:
heh this will probably be the first thing I use writing my next fn... just too easy to pass up 🙃